### DISCUSSION TITLE
Compile source code

### DISCUSSION THREAD

**Discussion by jredondo18** [neos-sdi/adfsmfa#269] [Q&A] [answered]
Hi redhook!

I am trying to compile the code, I have generated a .pfx key and pasted the solution.

I have also changed the PublicKey and PublicKeyToken values.

But I get a lot of compilation errors,

Do you have a more detailed guide to compile the project?

Thank you very much,

Regards.

**Comment by redhook62** [comment]
https://github.com/neos-sdi/adfsmfa/wiki/13-Build

Did you get all the nuget packages ?

Config files for nuget are present in the solution

regards

**Comment by jredondo18** [comment]
Hello again,

I think I already have the nuget packages. But when I start the project, I get the following three errors:

C:\Repos\adfsmfa\Neos.IdentityServer 3.1\Neos.IdentityServer.MultiFactor.WebAuthN.Core\Neos.IdentityServer.MultiFactor.WebAuthN.Core.csproj : error  : The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  C:\Repos\adfsmfa\Neos.IdentityServer 3.1\packages\System.Text.Json.6.0.6\build\System.Text.Json.targets

C:\Repos\adfsmfa\Neos.IdentityServer 3.1\Neos.IdentityServer.MultiFactor.WebAuthN.Provider\Neos.IdentityServer.MultiFactor.WebAuthN.Provider.csproj : error  : The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  C:\Repos\adfsmfa\Neos.IdentityServer 3.1\packages\System.Text.Json.6.0.6\build\System.Text.Json.targets

C:\Repos\adfsmfa\Neos.IdentityServer 3.1\Neos.IdentityServer.MultiFactor.WebAuthN.Core\Neos.IdentityServer.MultiFactor.WebAuthN.Core.csproj : error  : The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  C:\Repos\adfsmfa\Neos.IdentityServer 3.1\packages\System.Text.Json.6.0.6\build\System.Text.Json.targets

Do you know what they are? 

Thanks!

**Comment by redhook62** [comment, marked as answer]
Hi @jredondo18 

Listen ! it's still up to you to deal with this problem.

As mentioned, this project is compiled with VS 2017 up todate (no more needed), and it has been tested and works with VS 2019 see with @apr-un.

We don't support issues with other versions of Visual Studio. we remain for the moment on a source sharing and a build in VS2017.

Update your package manager, or rebuild a complete solution with the Visual Studio you're using.
It's not a source code issue, presumably you can compile with quite a few VS versions.

Really, we don't have time to deal with this type of problem.
We already provide a packaged solution and all the source code. we put this project on codeplex and then on Git thinking we had help from the community, and this is not the case, with a few rare exceptions that we really thank.

Lest us know !

regards
